### ISSUE TITLE
Children API total children inconsistencies

### ISSUE THREAD

**Issue by jvega190** [craftercms/craftercms#6992]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

When calling the `studio/api/2/content/{site}/children` API, there's an inconsistency with the total number of children.
For example, when calling the API without systemTypes parameter, you get the total of children considering the level-descriptor

### Steps to reproduce

Steps:

Calling the API with and without `systemTypes` parameter in the payload returns an inconsistent total number of children

Without systemTypes:
```
children: [child1, child2],
levelDescriptor: {...},
total: 3
```
But if systemTypes is sent in the payload
```
{"path":"/site/website/index.xml","limit":2,"systemTypes":["folder","page"]}
```
this is the response:
```
children: [child1, child2],
levelDescriptor: {...},
total: 2
```

The UI is expecting the descriptor to be considered in the total. So the response with systemTypes is causing issues.
Depending on the fix for this ticket, the UI needs to be updated.

**Comment by jvega190** [comment]
https://github.com/craftercms/craftercms/issues/6988 depends on this ticket.

**Comment by phuongnq** [comment]
CI:5178

**Comment by phuongnq** [comment]
This could be the same issue:
CI: 5178

1. Create a new site from Editorial BP
2. Create folder `/site/components/documents` with 3 children: `ceii`, `documents`, and `versions`
3. Update ui.xml to have the following widget:
```
<widget id="craftercms.components.PathNavigatorTree">
                    <configuration>
                        <id>Documents</id>
                        <label>Documents</label>
                        <icon id="@mui/icons-material/DescriptionOutlined"/>
                        <rootPath>/site/components/documents</rootPath>
                        <locale>en</locale>
                        <limit>200</limit>
                        <excludes>
                            <exclude>/site/components/documents/versions</exclude>
                        </excludes>
                    </configuration>
                </widget>
```
4. Browse the Documents in the sidebar and notice that the `studio/api/2/content/{site}/children` returns `3` but it is actually `2` since we excluded the `versions`. And there is a link with the text `...1 more item` in the sidebar that produces nothing on click.
![2024-07-08_11-34](https://github.com/craftercms/craftercms/assets/2996543/b0fa6fac-978c-4ef7-a23b-8c668b7d61fb)

**Comment by jvega190** [comment]
Verified fixed. The issue that @phuongnq mentioned is also fixed

**Comment by sumerjabri** [comment]
Reopening for 4.1

**Comment by jvega190** [comment]
Verified fixed in 4.1.x
